### PR TITLE
Part 6: Completing the tests for LTI 1.1 - Section 2

### PR DESCRIPTION
**Pull request by jon-betts**
This part of a series of changes for #1220, so you may see odd things which are in place for PRs later in the sequence.

This should be merged after #1236.

### CODE REVIEW

File: tests/bdd/steps/feature_steps/fixture.feature [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Feature: Fixture helpers
+  Scenario: the user is a {role}
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/fixture.feature]: There is a slight ambiguity here in terms of user ownership: is this a user per the LTI  request or our interpretation of an LTI request and a "user" on the LMS app's end? I believe it is the former, but the ambiguity makes a small confusion here.

File: tests/bdd/steps/feature_steps/fixture.feature [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    Given I load the fixture '{role}.ini' as 'role'
+      And I update the fixture 'params' from fixture 'role'
+
+  Scenario: the user is an {role}
+    Given the user is a {role}
+
+  Scenario: the request is for resource '{resource_id}'
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/fixture.feature]: _resource_ is a slightly vague term. Not that I have a better suggestion...

File: tests/bdd/steps/feature_steps/fixture.feature [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Given I load the fixture 'resource_{resource_id}.ini' as 'resource'
+      And I update the fixture 'params' from fixture 'resource'
+
+  Scenario: the params fixture matches the LTI example for section {test_name}
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/fixture.feature]: Scenario naming is a bit Byzantine here, but not the end of the world.

File: tests/bdd/steps/feature_steps/lms.feature [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Feature: Steps for our app
+  Scenario: the assigment opens successfully
+    Then the response is HTML
+    Then the response status code is 200
+
+  Scenario: the user has instructor privileges
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/lms.feature]: Now are we talking about an LMS-app user?

File: tests/bdd/steps/feature_steps/lms.feature [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    Then the response body does not match 'An instructor needs to launch the assignment to configure it.'
+
+  Scenario: the user only has learner privileges
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/lms.feature]: Ditto...

File: tests/bdd/steps/feature_steps/fixture.feature (hunk 2 above)
Comment (jon-betts) [reply tests/bdd/steps/feature_steps/fixture.feature]: The LTI spec is full of hideously vague terms, but I'm afraid that's what they call an assigment _some_ of the time. In this case it's something linked to via a `resource_link_id`. I honestly don't know if it's a `resource_link` or a `resource`. Both are super vague.

File: tests/bdd/steps/feature_steps/lms.feature (hunk 4 above)
Comment (jon-betts) [reply tests/bdd/steps/feature_steps/lms.feature]: The real person again.

**Pull request merged**